### ISSUE TITLE
E0001: Error DisableExecuteApiEndpoint works only within 'DefinitionBody' property

### ISSUE THREAD

**Issue by Stompf** [aws-cloudformation/cfn-lint#2585]
### CloudFormation Lint Version

0.73.1

### What operating system are you using?

Ubuntu

### Describe the bug

I want to disable the execute api endpoint in my RestAPI using SAM. This works as expected in deployment by adding the ` DisableExecuteApiEndpoint: True` property on `Properties` for a `AWS::Serverless::Api` type (see reproduction template). 

However cfn-lint gives this error message running:

```
$ cfn-lint -t template.yaml

E0001 Error transforming template: Resource with id [ApiGatewayRestApi] is invalid. DisableExecuteApiEndpoint works only within 'DefinitionBody' property.
template2.yaml:1:1
```

### Expected behavior

No E0001 lint error when using DisableExecuteApiEndpoint on a SAM `AWS::Serverless::Api`

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:
  ApiGatewayRestApi:
    Type: AWS::Serverless::Api
    Properties:
      OpenApiVersion: "3.0.3"
      DisableExecuteApiEndpoint: True
      StageName: default
      EndpointConfiguration:
        Type: REGIONAL

  Function:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs16.x
      InlineCode: |
        exports.handler = async (event) => ++event;
      Events:
        Notify:
          Type: Api
          Properties:
            Method: POST
            Path: /
            RestApiId: !Ref ApiGatewayRestApi
```

**Comment by kddejong** [comment]
This is coming from sam translator package.  I need to look into that package's logic to see if we need to alter the template to make it work correctly.

**Comment by kddejong** [comment]
https://github.com/aws/serverless-application-model/blob/83306d38ff957152f7f274eef3b9b9abd873101f/samtranslator/model/api/api_generator.py#L322-L325
